File: thrift_python/server/ServiceServer.py
import json

from thrift.protocol import TBinaryProtocol
from thrift.server import TServer
from thrift.transport import TSocket, TTransport

from server.genPy import DIYFrameworkService
from server.genPy.ttypes import Request
from server.handler.ServiceHandlerImpl import ServiceHandlerImpl

if __name__ == '__main__':
    serviceHandler = ServiceHandlerImpl()
    processor = DIYFrameworkService.Processor(serviceHandler)
    transport = TSocket.TServerSocket(host='127.0.0.1', port=9090)
    tfactory = TTransport.TBufferedTransportFactory()
    #tfactory = TTransport.TFramedTransportFactory()
    pfactory = TBinaryProtocol.TBinaryProtocolFactory()
    #server = TServer.TSimpleServer(processor, transport, tfactory, pfactory)
    server = TServer.TSimpleServer(processor, transport, tfactory, pfactory)
    print('Starting the server...')
    server.serve()

File: thrift_java/src/main/java/com/fdu/server/ServiceServer.java
package com.fdu.server;


import com.fdu.server.genJava.generalize.DIYFrameworkService;
import com.fdu.server.handler.ServiceHandlerImpl;
import org.apache.thrift.TProcessor;
import org.apache.thrift.server.TServer;
import org.apache.thrift.server.TSimpleServer;
import org.apache.thrift.transport.TServerSocket;
import org.apache.thrift.transport.TTransportException;

public class ServiceServer {
    /**
     * 启动thrift服务器
     */
    public static void main(String[] args) {
        try {
            System.out.println("服务端开启....");
            // 1.创建TProcessor
            TProcessor tprocessor = new DIYFrameworkService.Processor<DIYFrameworkService.Iface>(new ServiceHandlerImpl());
            TServerSocket serverTransport = new TServerSocket(9898);
            TServer server = new TSimpleServer(new TServer.Args(serverTransport).processor(tprocessor));
            server.serve();
//            TServer.Args tArgs = new TServer.Args(serverTransport);
//            tArgs.processor(tprocessor);
//            tArgs.protocolFactory(new TBinaryProtocol.Factory());
//            TServer server = new TSimpleServer(tArgs);
//            server.serve();
        } catch (TTransportException e) {
            e.printStackTrace();
        }
    }
}